client/src/main/scala/org/http4s/client/Timeout.scala

```diff
@@ -0,0 +1,50 @@
+package org.http4s
+package client
+
+import cats._
+import scala.concurrent.duration._
+
+/** ADT for the timeout types which may be used by a http4s client.
+  *
+  * Not all clients support all timeout types.
+  *
+  * See [[Timeouts]] for a mechanism to construct timeout values which always
+  * line up to expected constraints, e.g. responseHeaderTimeout < idleTimeout.
+  */
+sealed trait Timeout extends Product with Serializable {
```

Comment:
Is the value of these subtypes to prevent people calling them in the wrong order on `Timeouts`?

Reply:
Yes, and also it could be useful in other places in the API where we have raw `Duration` values in neighboring positions in parameters.